src/Semaphore/SemaphoreCharacter.tsx

```diff
@@ -0,0 +1,238 @@
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.East)}
+        />
+        <input
+          className="SemaphoreCharacter-southEast"
+          type="checkbox" 
+          checked={this.state.directions.indexOf(Direction.SouthEast) >= 0}
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.SouthEast)}
+        />
+        <input
+          className="SemaphoreCharacter-south"
+          type="checkbox" 
+          checked={this.state.directions.indexOf(Direction.South) >= 0}
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.South)}
+        />
+        <input
+          className="SemaphoreCharacter-southWest"
+          type="checkbox" 
+          checked={this.state.directions.indexOf(Direction.SouthWest) >= 0}
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.SouthWest)}
+        />
+        <input
+          className="SemaphoreCharacter-west"
+          type="checkbox" 
+          checked={this.state.directions.indexOf(Direction.West) >= 0}
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.West)}
+        />
+        <input
+          className="SemaphoreCharacter-northWest"
+          type="checkbox" 
+          checked={this.state.directions.indexOf(Direction.NorthWest) >= 0}
+          onChange={(event: React.FormEvent<HTMLInputElement>) => this.onChecked(event, Direction.NorthWest)}
+        />
+        <svg width="240" viewBox="0 0 76 64">
+          <defs>
+            <g id="Person" fill="black">
+              <circle r="3.5" cx="4" cy="4" />
+              <rect y="8" width="8" height="16" rx="2" ry="2" />
+            </g>
+            <g id="Flag">
+              <polyline strokeWidth="1" stroke="black" points="0 0, 0 20" />
+              <polyline strokeWidth="2" strokeLinecap="round" stroke="black" points="0 20, 0 28" />
+              <rect stroke="black" height="10" width="10" />
+              <polygon fill="red" points="0 0, 10 0, 0 10" />
+              <polygon fill="yellow" points="10 0, 10 10, 0 10" />
+            </g>
+            <g id="LeftFlag">
+              <use xlinkHref="#Flag" transform="scale(-1, 1)" />
+            </g>
+          </defs>
+          <use xlinkHref="#Person" x="34" y="23" />
+          {this.getLeftFlag()}
+          {this.getRightFlag()}
+          Sorry, your browser does not support inline SVG.
+        </svg>
+      </div>
+    );
+  }
+
+  private updateState() {
+    const { left, right } = SemaphoreCharacter.getAnglesFromDirections(this._directions);
+
+    this.setState({
+      left: left,
+      right: right,
+      directions: this._directions,
+    });
+  }
+
+  private getLeftFlag() {
+    if (this.state.left >= 0) {
+      return (
+        <use
+          transform={`rotate(${this.state.left}, 33, 32)`}
```

Comment:
What's 33,32 here - the central axis around which the rotation is performed?  Consider comment explaining the units here.

Reply:
It seems like it's the React transform rotate property.  Still trying to find where that syntax is documented since https://facebook.github.io/react-native/docs/transforms.html#transform doesn't seem to go into it...

Reply:
From some web searching, it looks like my suspicion that it's the center of rotation is correct.  Still may be worth a comment explaining the units and how they relate to the positioning of various elements here.